packages/gestalt/src/IconButtonFloating.js

```diff
@@ -0,0 +1,90 @@
+// @flow strict
+import { forwardRef, type Node } from 'react';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+import icons from './icons/index.js';
+
+type Props = {|
+  /**
+   * Specifies the `id` of an associated element (or elements) whose contents or visibility are controlled by IconButtonFLoating so that screen reader users can identify the relationship between elements. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityControls?: string,
+  /**
+   * Indicates that IconButtonFloating hides or exposes collapsible components and expose whether they are currently expanded or collapsed. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityExpanded?: boolean,
+  /**
+   * Indicates that this component controls the appearance of interactive popup elements, such as Dropdown or Modal. See the [Accessibility](#ARIA-attributes) guidelines for details on proper usage.
+   */
+  accessibilityHaspopup?: boolean,
+  /**
+   * String that clients such as VoiceOver will read to describe the icon button. Always localize the label. See [Accessibility section](https://gestalt.pinterest.systems/web/iconbuttonfloating#Accessibility) for more info.
+   */
+  accessibilityLabel: string,
+  /**
+   * Icon displayed in IconButtonFloating to convey the behavior of the component. Refer to the [iconography](/foundations/iconography/library) guidelines regarding the available icon options.
+   */
+  icon: $Keys<typeof icons>,
+  /**
+   * Callback fired when the component is clicked, pressed or tapped.
+   */
+  onClick?: ({|
```

Comment:
I think onClick should be requires as the main action for IconButton is to "open" a Modal and or Dropdown, onClick must be used to trigger that response.